server/models/User.js

```diff
@@ -35,6 +35,50 @@ class User extends Model {
             .send();
     }
 
+    async createResetLink() {
+        const token = crypto.randomBytes(20).toString('hex');
+        const date = new Date();
+        date.setHours(date.getHours() + 2);
+
+        await this.constructor
+            .knex()
+            .table('users')
+            .where('email', this.email)
+            .update({
+                resettoken: token,
+                resetexpire: date
+            });
+        await this.sendResetEmail(token);
+    }
+
+    async sendResetEmail(token) {
+        await new Mail('reset-password')
+            .to(this.email)
+            .subject('You requested for a password reset.')
+            .data({
+                name: this.firstName,
+                url: `${config.server.url}/api/v1/auth/reset/${token}`
+            })
+            .send();
+    }
+
+    async resetPassword(password) {
+        await this.$query().patch({
```

Comment:
Why two database queries  ? Why one for update password and another for `resettoken` and `resetexpire` ? Use one, most preferrably the patch.

Reply:
fixed